src/Data/PreciseDateTime.purs

```diff
@@ -180,7 +181,7 @@ diff (PreciseDateTime dt0 (Nanosecond ns0)) (PreciseDateTime dt1 (Nanosecond ns1
   let Duration.Milliseconds msNum = DateTime.diff dt0 dt1
       nsNum = Int.toNumber (ns0 - ns1) + msNum * 1000000.0
   in
-    Nanoseconds (BigInt.fromNumber nsNum)
+    PD.unsafeNanoseconds (Decimal.fromNumber nsNum)
```

Comment:
Does this need to be unsafe?

Reply:
Yes, because the number is already a `Decimal` and we're simply slapping a `Nanoseconds` label on it here